pages/static/survey.js

```diff
@@ -0,0 +1,255 @@
+const c = document.createElement.bind(document);
+const $ = document.querySelector.bind(document);
+const $$ = document.querySelectorAll.bind(document);
+
+const form = $('#fez');
+const dismiss = $('.ap-a-ico');
+
+const _safe = function (str) {
+  return str.replace(/ /g, '_').replace(/[^a-zA-Z]/g, '');
+};
+
+function Fez(data) {
+  if (data === null) {
+    throw new Error(
+      'Surveys must have an <amp-state> element, with the ID "surveyQuestions"'
+    );
+  }
+
+  try {
+    this.questions = this.validate(data);
+    this.name = data.survey;
+    this.url = data._url;
+    this.shownAt = new Date().toISOString();
+  } catch (e) {
+    throw new Error(e);
+  }
+
+  this.init();
+}
+
+Fez.prototype._questionTypes = [
+  'likert',
+  'multiple',
+  'open',
+  'rating',
+  'single',
+];
+
+Fez.prototype.validate = function (data) {
+  if (!Array.isArray(data.questions)) {
+    throw new Error('Survey data must include questions');
+  }
+
+  data.questions.forEach((q) => {
+    if (typeof q.text !== 'string') {
+      throw new Error('Survey questions must include a `text` field');
+    }
+    if (typeof q.type !== 'string') {
+      throw new Error('Survey questions must include a `type` field');
+    }
+    if (this._questionTypes.indexOf(q.type) === -1) {
+      throw new Error(q.type + ' is not a valid survey question type');
+    }
+  });
+
+  return data.questions;
+};
+
+Fez.prototype.init = function () {
+  const answerExpiration = localStorage.getItem(`survey_${this.name}`);
+  const surveyTimestamp = parseInt(answerExpiration);
+
+  // if the answer expired, or there is no saved answer
+  // (and is therefore a NaN when parsed)...
+  if (Number.isNaN(surveyTimestamp) || Date.now() > new Date(surveyTimestamp)) {
+    localStorage.removeItem(`survey_${this.name}`);
+
+    this.build();
+  }
+};
+
+Fez.prototype._attachListeners = function () {
+  const inputs = [...$$('[type=radio]'), ...$$('[type=button]')];
+
+  inputs.forEach((e) => {
+    e.addEventListener('click', (e) => {
+      const el = e.target;
+      let card = el;
+
+      while (card && !card.classList.contains('active')) {
+        card = card.parentNode;
+      }
+
+      const index = card.parentNode.children.indexOf(card);
+      const question = this.questions[index];
+
+      if (question.type === 'likert' || question.type === 'rating') {
+        question.answer = el.nextSibling.textContent;
```

Comment:
Would it make more sense to select by element type / Class? Otherwise HTML changes would break this.

Reply:
The template is generated in this file, so this style was my preference (as opposed to adding the className inline, and then calling that same string here). But whichever you prefer is fine by me